components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java

```diff
@@ -902,9 +902,21 @@ public long contentLength() throws IOException {
 
                                 @Override
                                 public void writeTo(@Nonnull BufferedSink sink) throws IOException {
+                                    long contentLength = contentLength();
+                                    if (contentLength > 0) {
+                                        requestInfo.content.mark((int) contentLength);
```

Comment:
why are we marking to the content length? isn't this effectively setting the stream to its end, making it look empty to the client when it comes time to write to the network?

Reply:
since we are calling mark() before the stream is read, we are setting the reading position to the beginning of the stream. 
The parameter passed to mark() affects the `reset()` by configuring how many bytes can be read from the stream before the mark is invalid

The parameter passed to `mark()` doesn't affect the start position of the read().

https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#mark-int-

The unit [tests](https://github.com/microsoft/kiota-java/blob/main/components/http/okHttp/src/test/java/com/microsoft/kiota/http/OkHttpRequestAdapterTest.java#L410-L536) confirm this.